File: csharp/src/Training.Workshop/M5/S15Compatibility/Step1/BoxOfficeApi.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M5.S15Compatibility.Step1;

/// <summary>Krok 1: bez zmian.</summary>
public sealed class BoxOfficeApi
{
    public Money Quote(StudentTicket ticket)
    {
        return ticket.Price();
    }
}

File: typescript/src/workshop/m5/s15_compatibility/step1/BoxOfficeApi.ts
import type { Money } from '../../../shared/Money.js';
import type { StudentTicket } from './StudentTicket.js';

/**
 * Krok 1: bez zmian.
 */
export class BoxOfficeApi {
  quoteStudent(ticket: StudentTicket): Money {
    return ticket.price();
  }
}
